kafka/consumer/main.go

```diff
@@ -151,37 +167,21 @@ func main() {
 				message = string(plaintext)
 			}
 			select {
+			case sig := <-signals:
+				log.Printf("Got signal: %v", sig)
+				return
 			case relayMessage <- message:
 			default:
 			}
 			log.Printf("Message received: %s\n", message)
 		}
-
-		select {
-		case sig := <-signals:
-			log.Printf("Got signal: %v", sig)
-			return
-		default:
-			log.Println("Have not reiceved signal yet. Continue processing..")
-		}
```

Comment:
You still need this.  If no events were received in the 10s timeout, you'll never do and signal handling.